backend/src/main/kotlin/org/darkSolace/muse/securityModule/controller/AuthController.kt

```diff
@@ -0,0 +1,63 @@
+package org.darkSolace.muse.securityModule.controller
+
+import org.darkSolace.muse.securityModule.model.LoginRequest
+import org.darkSolace.muse.securityModule.model.SignUpRequest
+import org.darkSolace.muse.securityModule.model.SignUpResponse
+import org.darkSolace.muse.securityModule.service.AuthenticationService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+
+@RestController
+@RequestMapping("api/auth")
+class AuthController(@Autowired val authenticationService: AuthenticationService) {
+    /**
+     * Checks a transmitted [LoginRequest] for a valid username/password pair. Listens on /api/auth/signin.
+     *
+     * @sample `curl -X POST -H "Content-Type: application/json" -d '{ "username": "test", "password": "123" }'
+     *          172.22.128.1:8000/api/auth/signin`
+     * @param loginRequest a [LoginRequest] containing username and password
+     * @return a [org.darkSolace.muse.securityModule.model.JwtResponse] containing
+     * a token or HTTP 401 is username or password are invalid
+     */
+    @PostMapping("/signin")
+    fun authenticateUser(@RequestBody loginRequest: LoginRequest): ResponseEntity<*>? {
+        val token = authenticationService.authenticate(loginRequest)
+        return if (token == null) {
+            ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Username or password wrong!")
```

Comment:
"Unknown username or wrong password!" ?
Or would this text never be displayed to the user and just be something for logging because the frontend deals with the HttpStatus?

Reply:
The message is handed back to the caller (in most cases the front-end), together with HTTP 401. If the message will be shown directly or a different message is chosen based on the HTTP Status Code is up to the caller. I changed it to your suggestion to allow to use this message directly within the front-end (and also changed the behavior when a user does not exist to deliver the same response - this was missing before).